From 0915b8b2531ccaada36639b12f7aafe152655143 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 9 Apr 2017 17:27:37 +0300
Subject: [PATCH 1582/2241] fix: net: mvpp2x: skip RX checksum offloading if
 feature disabled

Issue:
Rx checksum offloading was performed with disabled RXCSUM device feature.
Issue didn't caused any seen damage, but wrong skb ip_summed state
were set.

Fix:
Skip Rx checksum offloading if feature disabled.

Change-Id: I68111c66f837c185194ec45101ddf13bbb037ab0
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38628
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 917d103..7546af5 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -2425,7 +2425,10 @@ err_drop_frame:
 #endif
 		skb_put(skb, rx_bytes);
 		skb->protocol = eth_type_trans(skb, dev);
-		mv_pp2x_rx_csum(port, rx_status, skb);
+
+		if (likely(dev->features & NETIF_F_RXCSUM))
+			mv_pp2x_rx_csum(port, rx_status, skb);
+
 		skb_record_rx_queue(skb, (u16)rxq->log_id);
 		mv_pp2x_set_skb_hash(rx_desc, rx_status, skb);
 		skb_mark_napi_id(skb, napi);
-- 
2.7.4

